Adds example

M4tteoP opened the pull request:
This PR provides a quick deployment option in order to manually spin up an Envoy container with Coraza-proxy-wasm and the CRS enabled, working in Anomaly Scoring mode (once a certain threshold is reached, requests are going to be denied).

Note that:
- `example/tests.sh`: Has been temporarily added in order to provide immediately a first test that can be manually executed. Soon it is going to be replaced (and most likely deleted) in favor of an updated one that will be under `e2e/`
- WIP: Readme about example usage

jcchavezs commented on magefile.go:
```diff
@@ -185,4 +186,14 @@ func Ftw() error {
 }
 
+// Setup spins up the test environment. Requires docker-compose.
+func SetupExample() error {
```
Maybe `runExample` is more accurate?

jcchavezs commented:
Shall we run the example on CI?

anuraaga reviewed (commented):
Can we repurpose the e2e into an example? We already have it and ftw so this is yet another similar one. I think ftw is the main test anyways so we should be able to switch the e2e to an example?

M4tteoP commented:
The real and only difference was about embedding or not the CRS (e2e was intended to test basic Coraza phases without CRS). I agree about the overlap, working on merging e2e and example logic

M4tteoP commented:
Updated: I split the work opening [another PR](https://github.com/corazawaf/coraza-proxy-wasm/pull/19) to address the e2e refacor, It may require some specific conversations about response body.

jcchavezs commented:
Somehow github.com/tetratelabs/wazero@v1.0.0-beta.2 makes the linter work fine but github.com/tetratelabs/wazero@v1.0.0-pre.2 breaks it. It am not 100% sure it is the library itself or just a bad dependency resolution but let's defer this to a next PR.

For now I am opening a new PR to get this finally merged https://github.com/corazawaf/coraza-proxy-wasm/pull/30
